src/main/java/com/iri/dao3/dao/jdbc/DriverDaoImpl.java

```
@@ -0,0 +1,116 @@
+            Driver driver = null;
+            while (resultSet.next()) {
+                return constructDriverFromResultSet(resultSet);
+            }
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can't receive driver in method get", e);
+        }
+        return Optional.empty();
+    }
+
+    @Override
+    public List<Driver> getAll() {
+        String selectQuery = "SELECT * FROM drivers WHERE deleted = false";
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement preparedStatement = connection.prepareStatement(selectQuery)
+        ) {
+            ResultSet resultSet = preparedStatement.executeQuery();
+            List<Driver> drivers = new ArrayList<>();
+            while (resultSet.next()) {
+                drivers.add(constructDriverFromResultSet(resultSet).get());
+            }
+            return drivers;
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can't get connection in method getAll", e);
+        }
+    }
+
+    @Override
+    public Driver update(Driver driver) {
+        String updateQuery = "UPDATE drivers SET name = ?, license_number = ? "
+                + "WHERE id = ? AND deleted = false";
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement preparedStatement = connection.prepareStatement(updateQuery)
+        ) {
+            preparedStatement.setString(1, driver.getName());
+            preparedStatement.setString(2, driver.getLicenceNumber());
+            preparedStatement.setLong(3, driver.getId());
+            int result = preparedStatement.executeUpdate();
+            if (result > 0) {
+                return driver;
+            }
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can't get connection in method update driver", e);
+        }
+        throw new DataProcessingException("Can't update driver");
```

Should we throw an exception if nothing has been updated?

---

driver is unreachable